From 2edea35366278e35f3aff5ab854586a15dcec889 Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:38:55 -0500
Subject: [drm] radeon: don't reset dma on NI/SI init

Message-id: <1418852380-13061-45-git-send-email-rclark@redhat.com>
Patchwork-id: 102214
O-Subject: [RHEL7 drm 44/89] drm/radeon: don't reset dma on NI/SI init
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: since v3.16.4

commit 63dcd5a8b861b7b447dd49690f1b7a016f203739

Author:     Alex Deucher <alexander.deucher@amd.com>
AuthorDate: Wed Sep 17 17:41:04 2014 -0400
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:19:56 2014 -0500

    drm/radeon: don't reset dma on NI/SI init

    commit 31a25e2caf9367365fcb0e57fd8fa5a42e9b73e4 upstream.

    Otherwise we may lose the DMA golden settings which can
    lead to hangs, etc.

    bug:
    https://www.libreoffice.org/bugzilla/show_bug.cgi?id=83500

    Reviewed-by: Christian Koenig <christian.koenig@amd.com>
    Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Rob Clark <rclark@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/radeon/ni_dma.c b/drivers/gpu/drm/radeon/ni_dma.c
index 6378e02..6927db4 100644
--- a/drivers/gpu/drm/radeon/ni_dma.c
+++ b/drivers/gpu/drm/radeon/ni_dma.c
@@ -191,12 +191,6 @@ int cayman_dma_resume(struct radeon_device *rdev)
  u32 reg_offset, wb_offset;
  int i, r;
 
- /* Reset dma */
- WREG32(SRBM_SOFT_RESET, SOFT_RESET_DMA | SOFT_RESET_DMA1);
- RREG32(SRBM_SOFT_RESET);
- udelay(50);
- WREG32(SRBM_SOFT_RESET, 0);
-
  for (i = 0; i < 2; i++) {
   if (i == 0) {
    ring = &rdev->ring[R600_RING_TYPE_DMA_INDEX];
-- 
1.7.1